src/frontend/Semantic_check.ml

```diff
@@ -290,9 +290,30 @@ let semantic_check_fn_normal ~is_cond_dist ~loc id es =
     | Some _ ->
         (* Check that Funaps are actually functions *)
         Semantic_error.returning_fn_expected_nonfn_found loc id.name |> error
-    | None ->
-        Semantic_error.returning_fn_expected_undeclaredident_found loc id.name
-        |> error)
+    | None -> (
+      match List.rev (String.split id.name ~on:'_') with
+      | (("lupdf" | "cdf" | "lcdf" | "lccdf" | "rng") as suffix) :: tl ->
```

Comment:
This should probably be another constant in https://github.com/stan-dev/stanc3/blob/b0f2bd3de4e19ed9fb57e556cc4ba539af234195/src/middle/Utils.ml#L11

I guess none of the ones currently there are useful.

You can also skip `lupdf`. If there is an lpdf/lpmf implementation, that automatically means a lupdf/lupmf is available. If it weren't, we also couldn't use the lpdf/lpmf in a sampling statement (with the tilde).